File: abc/abc149/abc149_c.cs
// https://atcoder.jp/contests/abc149/submissions/9230269

using System;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var X = int.Parse(Console.ReadLine());

            int p = X;
            while (!IsPrime(p))
            {
                p++;
            }
            Console.WriteLine(p);
        }

        private static bool IsPrime(int x)
        {
            if (x <= 1)
            {
                return false;
            }
            for (int i = 2; i * i <= x; i++)
            {
                if (x % i == 0)
                {
                    return false;
                }
            }
            return true;
        }
    }
}

File: abc/abc149/abc149_c.fs
// https://atcoder.jp/contests/abc149/submissions/12184048

let X = stdin.ReadLine() |> int

let inline sqrtInt n =
    n
    |> float
    |> sqrt
    |> int

let isPrime x =
    let rec loop n =
        if n <= 1 then true
        elif x % n = 0 then false
        else loop (n - 1)
    loop (sqrtInt x)

let rec solver x =
    if isPrime x then x
    else solver (x + 1)

solver X |> stdout.WriteLine
